### ISSUE TITLE
No dedicated chart  for metrics in ClearML

### ISSUE THREAD

**Issue by darouwan** [ultralytics/ultralytics#21524]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I found in latest version, there is no dedicated chart for mAP50, mAP50-95, precision, recall, like this
<img width="830" height="428" alt="Image" src="https://github.com/user-attachments/assets/22f1bcaf-7ff6-4576-9fc4-00fbad25aab5" />

For instead, these values are integrated into this chart. Because the range of loss value and mAP value is different, the chart looks very strange in most of time. Is it in design or a bug? 

<img width="802" height="406" alt="Image" src="https://github.com/user-attachments/assets/23f1fb23-ddeb-40f2-921d-abf6dd42e27d" />

### Environment

```
Ultralytics 8.3.170  Python-3.9.7 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 3060, 12288MiB)
Setup complete  (16 CPUs, 63.9 GB RAM, 328.5/931.5 GB disk)
OS                  Windows-10-10.0.26100-SP0
Environment         Windows
Python              3.9.7
Install             git
Path                F:\python_project\ultralytics\ultralytics
RAM                 63.93 GB
Disk                328.5/931.5 GB
CPU                 AMD Ryzen 7 5800X 8-Core Processor
CPU count           16
GPU                 NVIDIA GeForce RTX 3060, 12288MiB
GPU count           1
CUDA                11.8
```

### Minimal Reproducible Example

```
    model = YOLO('yolo11n.pt')
    model.train(data="coco128.yaml", epochs=20, batch=32, project='projects/yolov8_coco128',
                workers=2,
                patience=50,
                visualize=True)
```

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @darouwan, thank you for your detailed report and for using Ultralytics 🚀! This is an automated response to help get you the fastest support possible. An Ultralytics engineer will review your issue and assist you soon.

We recommend reviewing our [Docs](https://docs.ultralytics.com/) for additional guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are answered there.

Since this is a 🐛 Bug Report and you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (thank you!), this will help our team investigate further.

If you’re exploring custom training or have more questions, please include any relevant dataset samples or logs, and make sure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for real-time help on [Discord](https://discord.com/invite/ultralytics) 🎧, or discuss on [Discourse](https://community.ultralytics.com/). You can also connect with others on our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

To ensure your issue is not already resolved, please upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in any of these up-to-date verified environments with all dependencies (including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your contribution and willingness to help by submitting a PR! An Ultralytics engineer will follow up with you as soon as possible.

**Comment by Y-T-G** [comment]
There hasn't been any recnt updates to ClearML integration

https://github.com/ultralytics/ultralytics/blob/main/ultralytics/utils/callbacks/clearml.py

Do you know which version works correctly?

**Comment by darouwan** [comment]
@Y-T-G I tested and it shole between 100 and 150

**Comment by darouwan** [comment]
should be between 100 and 106~

**Comment by glenn-jocher** [comment]
Thanks for narrowing down the version range! That's helpful - it suggests something changed in the ClearML callback implementation around v8.1.07+. 

For your PR, I'd recommend comparing the [ClearML callback](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/utils/callbacks/clearml.py) between v8.1.06 and current to identify what changed in the metrics logging. The issue likely stems from how the metrics are being grouped or reported to ClearML's scalar logging system.

**Comment by Y-T-G** [comment]
It logs the metrics under "val" here:
https://github.com/ultralytics/ultralytics/blob/18fee3e7778af501bc69a34c21d768add20e2e43/ultralytics/utils/callbacks/clearml.py#L108

But it has been like that for a long time.

**Comment by darouwan** [comment]
@Y-T-G I also find that, but sometimes it seems correct~~~

**Comment by darouwan** [comment]
@Y-T-G I get the cause. When I print the metrics value with 
```
        for k, v in trainer.metrics.items():
            print(k, v)
            task.get_logger().report_scalar("val", k, v, iteration=trainer.epoch)
```

I see 
```
metrics/precision(B) 0.64671
metrics/recall(B) 0.59784
metrics/mAP50(B) 0.67294
metrics/mAP50-95(B) 0.50695
val/box_loss 1.07555
val/cls_loss 1.01322
val/dfl_loss 1.11167
```

So the problem is in `trainer.metrics`

**Comment by Y-T-G** [comment]
Is it different between old version and new?

**Comment by darouwan** [comment]
@Y-T-G  Need some time to find the old version~~ But at least I have already known how to fix that. The fixed version would be like this, and I will submit the PR. 

<img width="1671" height="907" alt="Image" src="https://github.com/user-attachments/assets/d92cacb5-5e94-40c9-8e55-28c7733b8038" />

**Comment by darouwan** [comment]
I create the PR here https://github.com/ultralytics/ultralytics/pull/21549

**Comment by glenn-jocher** [comment]
Great debugging work! You correctly identified that the issue stems from all metrics being logged under the same "val" category, which mixes metrics with different scales (mAP values 0-1 vs losses that can be much larger). Separating the metrics by type in your PR will definitely improve the ClearML visualization clarity. Thanks for taking the initiative to both diagnose and fix this issue!
